app-misc/anki/anki-23.12.1.ebuild

```diff
@@ -0,0 +1,955 @@
+		$(python_gen_cond_dep '
+			dev-python/pytest[${PYTHON_USEDEP}]
+			dev-python/mock[${PYTHON_USEDEP}]')
+	)
+"
+
+QA_FLAGS_IGNORED="usr/bin/anki-sync-server
+	usr/lib/python.*/site-packages/anki/_rsbridge.so"
+
+DOC_CONTENTS="Users with add-ons that still rely on Anki's Qt5 GUI can either
+switch to ${CATEGORY}/${PN}[-qt6], or temporarily set the environment variable
+ENABLE_QT5_COMPAT to 1 to have Anki install the previous compatibility code.
+The latter option has additional runtime dependencies. Please take a look
+at this package's optional runtime features for a complete listing.
+\n\nIn an early 2024 update, ENABLE_QT5_COMPAT will be removed, so this is not
+a long-term solution.
+\n\nAnki's user manual is located online at https://docs.ankiweb.net/
+\nAnki's add-on developer manual is located online at
+https://addon-docs.ankiweb.net/
+"
+
+pkg_setup() {
+	export PROTOC_BINARY="${BROOT}"/usr/bin/protoc
+	export ZSTD_SYS_USE_PKG_CONFIG=1
+
+	if use gui; then
+		python-single-r1_pkg_setup
+		export NODE_BINARY="${BROOT}"/usr/bin/node
+		if ! use debug; then
+			if tc-is-lto; then
+				export RELEASE=2
+			else
+				export RELEASE=1
+			fi
+		fi
+	fi
+}
+
+src_prepare() {
+	default
+	rm -r ftl/{core,qt}-repo || die
+	ln -s "${WORKDIR}"/anki-core-i18n-${COMMITS[ftl-core]} ftl/core-repo || die
+	ln -s "${WORKDIR}"/anki-desktop-ftl-${COMMITS[ftl-desktop]} ftl/qt-repo || die
+
+	mkdir out || die
+	echo -e "${COMMITS[anki]:0:8}" > out/buildhash || die
+	if use gui; then
+		mv "${WORKDIR}"/node_modules out || die
+
+		# Some parts of the runner build system expect to be in a git repository
+		mkdir .git || die
+
+		# Creating the pseudo venv early skips pip dependency checks in src_compile.
+		# out/pyenv/bin/python provides the expected output of the pyenv:bin target.
+		mkdir -p out/pyenv/bin || die
+		ln -s "${PYTHON}" out/pyenv/bin/"${EPYTHON}" || die
+		ln -s "${EPYTHON}" out/pyenv/bin/python || die
+		# TODO: ln -s "${BROOT}/usr/bin/protoc-gen-mypy" out/pyenv/bin || die
+
+		# Anki's Qt detection mechanism falls back to Qt5 Python bindings, if Qt6
+		# Python bindings don't get imported successfully.
+		if ! use qt6; then
+			sed -i "s/import PyQt6/raise ImportError/" qt/aqt/qt/__init__.py || die
+		fi
+	fi
+}
+
+src_compile() {
+	filter-lto
+	tc-export AR CC PKG_CONFIG
+	export CARGO_TARGET_DIR="${S}"/out/rust
+
+	# Overview of the build process
+	#
+	# 1. The "runner" build system is built explicitly with the "--release" flag
+	#     to avoid issues with hardcoded paths.  Once completed the binaries are
+	#     placed into "${S}"/out/rust/release.
+	# 2. As soon as step 1 is finished, the "runner" build system is executed,
+	#    which then processes the following steps:
+	#    * build remaining components of "runner" (= ninja_gen, configure, etc.)
+	#    * generate the ninja file and run ninja afterwards
+	#    * create the Python wheel files in "${S}"/out/wheels
+
+	cargo_src_compile $(usev debug "--release") --package runner
```

Comment:
Is is acceptable to explicitly set the `--release` flag this way? Unfortunately, rustc complains about it being set more than once.

Reply:
Do you mean:
```suggestion
	cargo_src_compile $(usev !debug "--release") --package runner
```
?

Reply:
No, I mean to run:
```suggestion
	cargo build --release --package runner || die
```

Reply:
Huh, okay.